community/cypher/acceptance/src/test/scala/org/neo4j/internal/cypher/acceptance/InQueryProcedureCallAcceptanceTest.scala

```diff
@@ -0,0 +1,254 @@
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.internal.cypher.acceptance
+
+import java.util
+
+import org.neo4j.cypher._
+import org.neo4j.kernel.api.proc.Neo4jTypes
+
+class InQueryProcedureCallAcceptanceTest extends ProcedureCallAcceptanceTest {
+
+  test("should be able to find labels from built-in-procedure") {
+    // Given
+    createLabeledNode("A")
+    createLabeledNode("B")
+    createLabeledNode("C")
+
+    //When
+    val result = execute("CALL db.labels() YIELD label RETURN *")
+
+    // Then
+    result.toList should equal(
+      List(
+        Map("label" -> "A"),
+        Map("label" -> "B"),
+        Map("label" -> "C")))
+  }
+
+  test("should be able to find labels from built-in-procedure from within a query") {
+    // Given
+    createLabeledNode(Map("name" -> "Tic"), "A")
+    createLabeledNode(Map("name" -> "Tac"), "B")
+    createLabeledNode(Map("name" -> "Toc"), "C")
+
+    //When
+    val result = execute("MATCH (n {name: 'Toc'}) WITH n.name AS name CALL db.labels() YIELD label RETURN *")
+
+    // Then
+    result.toList should equal(
+      List(
+        Map("name" -> "Toc", "label" -> "A"),
+        Map("name" -> "Toc", "label" -> "B"),
+        Map("name" -> "Toc", "label" -> "C")))
+  }
+
+  test("db.labels works on an empty database") {
+    // Given an empty database
+    //When
+    val result = execute("CALL db.labels() YIELD label RETURN *")
+
+    // Then
+    result.toList shouldBe empty
+  }
+
+  test("should be able to call procedure with explicit arguments") {
+    // Given
+    registerDummyInOutProcedure(Neo4jTypes.NTString, Neo4jTypes.NTNumber)
+
+    // When
+    val result = execute("CALL my.first.proc('42', 42) YIELD out0, out1 RETURN *")
+
+    // Then
+    result.toList should equal(List(Map("out0" -> "42", "out1" -> 42)))
+  }
+
+  test("should be able to call empty procedure") {
+    //Given
+    registerProcedureReturningNoRowsOrColumns()
+
+    //When
+    val result = execute("CALL sys.return_nothing() RETURN 1")
+
+    // Then
+    result.toList shouldBe empty
+  }
+
+  test("should be able to call void procedure") {
+    //Given
+    registerVoidProcedure()
+
+    //When
+    val result = execute("MATCH (n) CALL sys.do_nothing() RETURN n")
+
+    // Then
+    result.toList shouldBe empty
+  }
+
+  test("should be able to call void procedure without swallowing rows") {
+    //Given
+    registerVoidProcedure()
+
+    createLabeledNode("A")
+    createLabeledNode("B")
+    createLabeledNode("C")
+
+    //When
+    val result = execute("MATCH (n) CALL sys.do_nothing() RETURN n")
+
+    // Then
+    result.toList.size shouldBe 3
+  }
+
+  test("should fail to shadow already bound identifier from a built-in-procedure") {
+    // Given
+    createLabeledNode("A")
+    createLabeledNode("B")
+    createLabeledNode("C")
+
+    //When
+    a [SyntaxException] shouldBe thrownBy(
```

Comment:
Yeah we need to fix that, everything that comes out of semantic checking is a syntax error today.